modules/miner/persist.go

```diff
@@ -65,15 +68,35 @@ func (m *Miner) initPersist() error {
 
 // load loads the miner persistence from disk.
 func (m *Miner) load() error {
-	return persist.LoadFile(settingsMetadata, &m.persist, filepath.Join(m.persistDir, settingsFile))
-}
-
-// save saves the miner persistence to disk.
-func (m *Miner) save() error {
-	return persist.SaveFile(settingsMetadata, m.persist, filepath.Join(m.persistDir, settingsFile))
+	return persist.LoadJSON(settingsMetadata, &m.persist, filepath.Join(m.persistDir, settingsFile))
 }
 
 // saveSync saves the miner persistence to disk, and then syncs to disk.
 func (m *Miner) saveSync() error {
-	return persist.SaveFileSync(settingsMetadata, m.persist, filepath.Join(m.persistDir, settingsFile))
+	return persist.SaveJSON(settingsMetadata, m.persist, filepath.Join(m.persistDir, settingsFile))
+}
+
+// threadedSaveLoop periodically saves the miner persist.
+func (m *Miner) threadedSaveLoop() {
+	for {
+		select {
+		case <-m.tg.StopChan():
+			return
+		case <-time.After(saveLoopPeriod):
+		}
+
+		func() {
+			err := m.tg.Add()
+			if err != nil {
+				return
+			}
+			defer m.tg.Done()
+
+			m.mu.Lock()
+			err = m.saveSync()
+			if err != nil {
+				m.log.Println("ERROR: Unable to save miner persist:", err)
+			}
```

Comment:
Looks like you are right. Wish there were static tools to detect this type of mistake